### PR TITLE
Only list entities which do not have factories

### LINKED ISSUE #82: make:factory only list entities that don't have factories already
Related to #15

### PR DESCRIPTION
**Pull request by jschaedl**
Fixes #82

### CODE REVIEW

File: src/Bundle/Resources/skeleton/Factory.tpl.php [hunk 1]
```diff
@@ -50,7 +50,7 @@ protected function initialize(): self
         ;
     }
 
-    protected static function getClass(): string
+    public static function getClass(): string
```
Comment (jschaedl) [inline src/Bundle/Resources/skeleton/Factory.tpl.php:53]: Not happy about that change. Maybe you have a better idea how we can access the entities class name?

File: src/ModelFactory.php [hunk 2]
```diff
@@ -240,7 +240,7 @@ final protected function addState($attributes = []): self
     }
 
     /** @psalm-return class-string<TModel> */
-    abstract protected static function getClass(): string;
+    abstract public static function getClass(): string;
```
Comment (jschaedl) [inline src/ModelFactory.php:243]: Not happy about that change. Maybe you have a better idea how to access the entities class names?

Comment (kbond) [reply src/ModelFactory.php]: This is also a BC break.

It's kind of crappy but the only thing I can think of is adding a separate public static method that just calls the protected one. Marked as final and `@internal`.

**Review by kbond — CHANGES_REQUESTED** [review]
Let's just get the CI passing and I'll merge!

**Comment by jschaedl** [comment]
I still need to fix the tests.

**Comment by kbond** [comment]
I fixed the tests and added an assertion to ensure existing factory entities aren't displayed.

Thanks @jschaedl!

**Review by kbond — APPROVED** [review]

**Comment by wouterj** [comment]
Wohoo, this is great. Thanks @jschaedl!

**Pull request merged**